src/modules/auth/auth.service.ts

```diff
@@ -1,6 +1,85 @@
+import * as bcrypt from 'bcryptjs';
 import { Injectable } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { CreateUserInput } from '../user/dto/new-user.input';
+import { UserService } from '../user/user.service';
+import { Result } from '@/common/dto/result.type';
+import * as dotenv from 'dotenv';
+dotenv.config();
+
+import {
+  ACCOUNT_EXIST,
+  ACCOUNT_NOT_EXIST,
+  LOGIN_ERROR,
+  NOT_EMPTY,
+  REGISTER_ERROR,
+  SUCCESS,
+} from '@/common/constants/code';
 
 @Injectable()
 export class AuthService {
-  constructor() {}
+  constructor(
+    private userService: UserService,
+    private jwtService: JwtService,
+  ) { }
+
+  async login(email: string, password: string): Promise<Result> {
+    const user = await this.userService.findByEmail(email);
+    if (!user) {
+      return {
+        code: ACCOUNT_NOT_EXIST,
+        message: "account doesn't exist",
+      };
+    }
+    const isPasswordValid = await bcrypt.compare(password, user.password);
+    if (isPasswordValid) {
+      const token = this.jwtService.sign({ id: user.id });
+      return {
+        code: SUCCESS,
+        message: 'login successful',
+        data: token,
+      };
+    }
+    return {
+      code: LOGIN_ERROR,
+      message: 'login failed, wrong password',
+    };
+  }
+
+  async register(input: CreateUserInput): Promise<Result> {
+    if (!input.displayName) {
+      input.displayName =
+        process.env.DEFAULT_DISPLAY_NAME || 'Default_Display_Name';
+    }
+
+    if (!input.email || !input.password || !input.ref || !input.displayName) {
```

Comment:
since `ValidationPipe` is introduced in resolver, the logic of `register` service need no change?